### PR TITLE
Overhaul of 'Understanding Serverless' section

### PR DESCRIPTION
**Pull request by abrennan89**
Updates based on minimalism session feedback
- titles updated
- content reviewed, fluff removed, organised into sub-headings
- updated link in release notes and topic yaml info

### CODE REVIEW

File: serverless/serverless-getting-started.adoc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[id="serverless-getting-started"]
+= Getting started with {ServerlessProductName}
+include::modules/common-attributes.adoc[]
+include::modules/serverless-document-attributes.adoc[]
+:context: serverless-getting-started
+
+:FeatureName: OpenShift Serverless
+include::modules/technology-preview.adoc[leveloffset=+1]
+
+{ServerlessProductName} makes it simpler to get code from development into production by reducing the need for infrastructure set up or back-end development by developers.
```
Comment (ahardin-rh) [inline serverless/serverless-getting-started.adoc]: Consider rewording this a bit to help with translation:
{ServerlessProductName} simplifies the process of delivering code from development into production 
(or something like that  :thinking: )

File: serverless/serverless-getting-started.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+[id="how-it-works_{context}"]
+== How does {ServerlessProductName} work?
```
Comment (ahardin-rh) [inline serverless/serverless-getting-started.adoc]: Consider reformatting as a statement vs. question: 
How {ServerlessProductName} works

File: serverless/serverless-getting-started.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Developers on {ServerlessProductName} can use the provided Kubernetes-native APIs, as well as familiar languages and frameworks, to deploy applications and container workloads.
+
+For information about installing {ServerlessProductName}, see the xref:../serverless/installing-openshift-serverless.adoc#installing-openshift-serverless[Installing OpenShift Serverless] documentation.
```
Comment (ahardin-rh) [inline serverless/serverless-getting-started.adoc]: see xref:../serverless/installing-openshift-serverless.adoc#installing-openshift-serverless[Installing OpenShift Serverless].

File: serverless/serverless-getting-started.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+{ServerlessProductName} on {product-title} enables stateful, stateless, and serverless workloads to all run on a single multi-cloud container platform with automated operations.
+Developers can use a single platform for hosting their microservices, legacy, and serverless applications.
```
Comment (ahardin-rh) [inline serverless/serverless-getting-started.adoc]: Looks like an unnecessary new line?

**Comment by ahardin-rh** [comment]
@abrennan89 Just a few minor comments from me. Otherwise, looks great!

**Comment by abrennan89** [comment]
Thanks so much, @ahardin-rh :slightly_smiling_face: should be ready to merge now. This applies for 4.1+

**Pull request merged**